pkg/optimizely/cache.go

```diff
@@ -34,23 +34,25 @@ import (
 // OptlyCache implements the Cache interface backed by a concurrent map.
 // The default OptlyClient lookup is based on supplied configuration via env variables.
 type OptlyCache struct {
-	loader          func(string, config.ClientConfig, *MetricsRegistry) (*OptlyClient, error)
-	optlyMap        cmap.ConcurrentMap
-	ctx             context.Context
-	wg              sync.WaitGroup
-	clientConf      config.ClientConfig
-	metricsRegistry *MetricsRegistry
+	loader   func(string) (*OptlyClient, error)
+	optlyMap cmap.ConcurrentMap
+	ctx      context.Context
+	wg       sync.WaitGroup
 }
 
 // NewCache returns a new implementation of OptlyCache interface backed by a concurrent map.
 func NewCache(ctx context.Context, conf config.ClientConfig, metricsRegistry *MetricsRegistry) *OptlyCache {
+
+	// TODO is there a cleaner way to handle this translation???
+	cmLoader := func(sdkkey string, options ...sdkconfig.OptionFunc) SyncedConfigManager {
+		return sdkconfig.NewPollingProjectConfigManager(sdkkey, options...)
+	}
+
 	cache := &OptlyCache{
-		ctx:             ctx,
-		wg:              sync.WaitGroup{},
-		loader:          initOptlyClient,
-		optlyMap:        cmap.New(),
-		clientConf:      conf,
-		metricsRegistry: metricsRegistry,
+		ctx:      ctx,
+		wg:       sync.WaitGroup{},
+		loader:   defaultLoader(conf, metricsRegistry, cmLoader, event.NewBatchEventProcessor),
```

Comment:
Re, the TODO comment, what happens if we do this?
```suggestion
		loader:   defaultLoader(conf, metricsRegistry, sdkconfig.NewPollingProjectConfigManager, event.NewBatchEventProcessor),
```

Reply:
That's what I want to do, but that throws a compiler error. The method above coerces into the interface.